File: compiler/tests/bench_medium.go
package main

type Point struct {
	X float64
	Y float64
}

func distance(a, b Point) float64 {
	dx := a.X - b.X
	dy := a.Y - b.Y
	return dx*dx + dy*dy
}

func compute(n int) float64 {
	var total float64
	p1 := Point{1.0, 2.0}
	p2 := Point{4.0, 6.0}
	for i := 0; i < n; i++ {
		total += distance(p1, p2)
	}
	return total
}

func main() {
	_ = compute(1000)
}

File: compiler/tests/bench_medium.rs
struct Point {
    x: f64,
    y: f64,
}

fn distance(a: &Point, b: &Point) -> f64 {
    let dx = a.x - b.x;
    let dy = a.y - b.y;
    dx * dx + dy * dy
}

fn compute(n: i32) -> f64 {
    let mut total: f64 = 0.0;
    let p1 = Point { x: 1.0, y: 2.0 };
    let p2 = Point { x: 4.0, y: 6.0 };
    for _ in 0..n {
        total += distance(&p1, &p2);
    }
    total
}

fn main() {
    let _result = compute(1000);
}
